lldb/source/Target/TargetList.cpp

```diff
@@ -92,6 +92,15 @@ Status TargetList::CreateTargetInternal(
   ArchSpec platform_arch(arch);
 
   bool prefer_platform_arch = false;
+  auto update_platform_arch = [&](const ArchSpec &module_arch) {
+    // If the OS or vendor weren't specified, then adopt the module's
+    // architecture so that the platform matching can be more accurate.
+    if (!platform_arch.TripleOSWasSpecified() ||
+        !platform_arch.TripleVendorWasSpecified()) {
```

Comment:
Is this check strict enough? I thought it should be only `TripleOSWasSpecified` -- what we can infer from the vendor?

Reply:
Short answer: I don't know. To be honest, I think the entire function needs a specification followed by a rewrite. The purpose of this patch is merely to make fat binary loading behave like loading a sing-arch binary, and that's the code from the single-arch branch.